agent/conversion_service.py

```
@@ -0,0 +1,312 @@
+            amount: Amount to swap in token's smallest unit
+            slippage: Maximum slippage percentage
+        
+        Returns:
+            Tuple of (transaction_hash, transaction_receipt)
+        
+        Raises:
+            ValueError: If conversion account not configured
+            Exception: If swap execution fails
+        """
+        if not self.conversion_account:
+            raise ValueError("Conversion private key must be configured to execute swaps")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)